### PR TITLE
SOF-7142: use cypress base from TeDe

### PR DESCRIPTION
**Pull request by k0stik**

### CODE REVIEW

File: tests/cypress/support/step_definitions/I add boundary conditions with the following data.ts [hunk 1]
```diff
@@ -1,10 +1,17 @@
...
-import { parseTable } from "../utils/table";
+// import SETTINGS from "../settings";
 import { BoundaryConditions } from "../widgets/BoundaryConditionsDialogWidget";
-import { materialDesignerPage } from "../widgets/MaterialDesignerPage";
+import MaterialDesignerPage from "../widgets/MaterialDesignerPage";
+// import SETTINGS from "./settings";
+
+// beforeEach(() => {
+//     BrowserFactory.setBrowserSettings(SETTINGS);
+// });
```
Comment (timurbazhirov) [inline tests/cypress/support/step_definitions/I add boundary conditions with the following data.ts]: Pls clean

File: tests/cypress/support/widgets/Page.ts [hunk 2]
```diff
@@ -1,10 +1,5 @@
...
+import BasePage from "@exabyte-io/code.js/dist/cypress/Page";
 
-export default class Page extends Widget {
-    open() {
-        browser.go("/");
-        this.waitForVisible();
-        this.waitForLoaderToDisappear();
-    }
+export default class Page extends BasePage {
+    // Put custom actions here
```
Comment (timurbazhirov) [inline tests/cypress/support/widgets/Page.ts]: Custom functions?

File: tests/cypress/support/e2e.ts [hunk 3]
```diff
@@ -12,9 +12,13 @@
...
 // Import commands.js using ES2015 syntax:
 import "./commands";
 
-// Alternatively you can use CommonJS syntax:
-// require('./commands')
+import BrowserFactory from "@exabyte-io/code.js/dist/cypress/BrowserFactory";
+
+import SETTINGS from "./settings";
+
+beforeEach(() => {
```
Comment (timurbazhirov) [inline tests/cypress/support/e2e.ts]: before?

File: tests/cypress/support/step_definitions/I add boundary conditions with the following data.ts (hunk 1 above)
Comment (k0stik) [reply tests/cypress/support/step_definitions/I add boundary conditions with the following data.ts]: Done

File: tests/cypress/support/e2e.ts (hunk 3 above)
Comment (k0stik) [reply tests/cypress/support/e2e.ts]: Done

File: tests/cypress/support/widgets/Page.ts (hunk 2 above)
Comment (k0stik) [reply tests/cypress/support/widgets/Page.ts]: Done

File: tsconfig.json [hunk 4]
```diff
@@ -24,6 +24,7 @@
     "allowSyntheticDefaultImports": true,
     "noFallthroughCasesInSwitch": true,
     "resolveJsonModule": true,
+    "noEmit": true
```
Comment (VsevolodX) [inline tsconfig.json]: I see this thing `noEmit: true` appearing everywhere, that prevents MD from being started on Docker and Netlify. It's created by `create-react-app` while running `start` script. 
I wonder, were you able to run MD in Docker locally?

**Review by VsevolodX — COMMENTED** [review]
No problems to my eyes.

File: tsconfig.json (hunk 4 above)
Comment (timurbazhirov) [reply tsconfig.json]: Let's not commit, there should be a workaround: https://github.com/Exabyte-io/materials-designer/commit/7069296c51125075769527d79cb68d53945ae36e

**Review by timurbazhirov — APPROVED** [review]
